src/plugins/intel_gpu/src/runtime/format.cpp

```diff
@@ -190,4 +192,91 @@ std::string format::to_string() const {
     return traits(value).str;
 }
 
+format format::get_default_format(size_t rank, bool is_weights, bool is_grouped) {
+    auto default_fmt = cldnn::format::bfyx;
+    if (is_weights) {
+        if (is_grouped) {
+            if (rank == 5) {
+                default_fmt = cldnn::format::goiyx;
+            } else if (rank == 6) {
+                default_fmt = cldnn::format::goizyx;
+            }
+        } else {
+            if (rank == 4) {
+                default_fmt = cldnn::format::oiyx;
+            } else if (rank == 5) {
+                default_fmt = cldnn::format::oizyx;
+            }
+        }
+    } else {
+        if (rank == 5) {
+            default_fmt = cldnn::format::bfzyx;
+        } else if (rank == 6) {
+            default_fmt = cldnn::format::bfwzyx;
+        }
+    }
+    return default_fmt;
+}
+
+format format::adjust_to_rank(format fmt, size_t new_rank) {
```

Comment:
In general the logic looks good to me, but there is the chance of incorrect behavior in some cases because of uninformative descriptors (let's say b_fs_yx_32fp may be adjusted to bfzyx), so maybe it makes sense to reduce adjusting-friendly formats list...

Reply:
good catch with binary format. Added check for that.